sc2_controller/src/game/game_result.rs

```diff
@@ -0,0 +1,309 @@
+                    self.player1_result = Some(player_result);
+                }
+                PlayerNum::Two => {
+                    self.player2_result = Some(player_result);
+                }
+            }
+        }
+    }
+}
+
+impl From<&GameResult> for AiArenaGameResult {
+    fn from(game_result: &GameResult) -> Self {
+        let mut game_steps = 0;
+        let mut bot1_avg_step_time = None;
+        let mut bot1_tags = None;
+        let mut bot2_avg_step_time = None;
+        let mut bot2_tags = None;
+        let mut p1_result = None;
+        let mut p2_result = None;
+
+        match &game_result.player1_result {
+            None => {}
+            Some(player1_result) => {
+                debug!("Player1Result: {:?}", player1_result);
+                bot1_avg_step_time = Some(player1_result.frame_time);
+                bot1_tags = Some(player1_result.tags.iter().cloned().collect());
+                game_steps = player1_result.game_loops;
+                p1_result = Some(player1_result.result);
+            }
+        }
+        match &game_result.player2_result {
+            None => {}
+            Some(player2_result) => {
+                debug!("Player2Result: {:?}", player2_result);
+                bot2_avg_step_time = Some(player2_result.frame_time);
+                bot2_tags = Some(player2_result.tags.iter().cloned().collect());
+                game_steps = player2_result.game_loops;
+                p2_result = Some(player2_result.result);
+            }
+        }
+        let result = game_result
+            .result
+            .unwrap_or_else(|| match (p1_result, p2_result) {
+                (Some(Sc2Result::SC2Crash), _) | (_, Some(Sc2Result::SC2Crash)) => {
+                    AiArenaResult::Error
+                }
+                (Some(Sc2Result::Tie), _) | (_, Some(Sc2Result::Tie)) => AiArenaResult::Tie,
+                (Some(Sc2Result::Crash), _) => AiArenaResult::Player1Crash,
+                (_, Some(Sc2Result::Crash)) => AiArenaResult::Player2Crash,
+                (Some(Sc2Result::Timeout), _) => AiArenaResult::Player1TimeOut,
+                (_, Some(Sc2Result::Timeout)) => AiArenaResult::Player2TimeOut,
+                (Some(Sc2Result::Victory), _) | (_, Some(Sc2Result::Defeat)) => {
+                    AiArenaResult::Player1Win
+                }
+                (_, Some(Sc2Result::Victory)) | (Some(Sc2Result::Defeat), _) => {
+                    AiArenaResult::Player2Win
+                }
+                #[cfg(test)]
+                (Some(Sc2Result::Placeholder), Some(Sc2Result::Placeholder)) => unreachable!(),
+                (_, _) => unreachable!(),
+            });
+        Self {
+            match_id: game_result.match_id,
+            bot1_avg_step_time,
+            bot1_tags,
+            bot2_avg_step_time,
+            bot2_tags,
+            result,
+            game_steps,
+        }
+    }
+}
+
+pub static GAME_RESULT: Lazy<Arc<RwLock<GameResult>>> =
```

Comment:
I expect there's probably no easy way to avoid this global? I note also the `PLAYER_SEATS` one. If it's too painful - all good.